models/Result.js

```diff
@@ -280,78 +133,38 @@ class Result {
         });
         const filteredResultsIds = filteredResults.sort(resultsSort).map(({ id: resultId }) => resultId);
 
-        const switchQuery = client.query(switchSql, [filteredResultsIds]);
-
-        switchQuery.on('end', () => {
-          const oneUserQuery = client.query(oneUserSql, [onePersonResultIds]);
-
-          oneUserQuery.on('end', () => {
-            client.end();
-            resolve();
-          });
-          oneUserQuery.on('error', err => {
-            client.end();
-            logger.errorLog('Results.switchSpotsForPerformance: oneUserQuery', err);
-            reject(err);
-          });
-        });
-
-        switchQuery.on('error', err => {
-          client.end();
-          logger.errorLog('Results.switchSpotsForPerformance: switchQuery', err);
-          reject(err);
-        });
-      });
-
-      resultsQuery.on('error', err => {
-        logger.errorLog('Results.switchSpotsForPerformance', err);
-        client.end();
-        reject(err);
-      });
+        db.query(switchSql, [filteredResultsIds])
+        .then(() => {
+          db.query(oneUserSql, [onePersonResultIds])
+          .then(resolve)
+          .catch(reject);
+        })
+        .catch(reject);
+      })
+      .catch(reject);
```

Comment:
I'm not in a good enough state to unwrap those nested promises and that's a pretty important part of the code base. I'm going to save that one for later 😜